internal/statestorage/redis/playerindices/playerindices.go

```diff
@@ -0,0 +1,273 @@
+	"github.com/gomodule/redigo/redis"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	"github.com/tidwall/gjson"
+)
+
+var (
+	// Logrus structured logging setup
+	piLogFields = log.Fields{
+		"app":       "openmatch",
+		"component": "statestorage",
+	}
+	piLog = log.WithFields(piLogFields)
+
+	// OM Internal metadata indices
+	MetaIndices = []string{
+		"OM_METADATA.created",
+		"OM_METADATA.accessed",
+	}
+)
+
+// Indexing  is all done using Sorted Sets in Redis, which require integer
+// 'scores' for each attribute in order to index players.
+//
+// Here are the guidelines if you want to index a player attribute in your
+// Properties JSON blob when the  player's request comes in the Frontend  API
+// so that you can filter on that  attribute in the Profiles you pass to the
+// Backend API.
+//  - Fields you want to index in your JSON should always be a key with an
+//    integer value, so use dictionaries/maps/hashes instad of lists/arrays for
+//    your data. (see examples below)
+//  - When indexing fields in a player's JSON object, the key to index should
+//    be compatible with dot notation. This means no keys with meta characters
+//    (anything escaped by regexp.QuoteMeta() shouldn't be used in your key
+//    name!)
+//  - If you're trying to index a flag, just use the epoch timestamp of the
+//    request as the value unless you have a compelling reason to do otherwise.
+//
+// For example, if you want to index the following:
+//    - Player's ping value to us-east
+//    - Bool flag 'true' denoting player's choice to play CTF mode
+//    - Bool flag 'true' denoting player's choice to play TeamDM mode
+//    - Bool flag 'true' denoting player's choice to play SunsetValley map
+//    - Players' matchmaking ranking value
+//
+// DON'T structure your JSON like this:
+//   player {
+//     "pings": {"us-east": 70, "eu-central": 120 },
+//     "maps":  ["sunsetvalley", "bigskymountain"] ,
+//     "modes": "ctf"
+//   }
+// Instead, use dictionaries with key/value pairs instead of lists (use epoch
+// timestamp as the value if your attribute should act as a boolean flag):
+//   player {
+//     "pings": {"us-east": 70, "eu-central": 120 },
+//     "maps":  {"sunsetvalley": 1234567890, "bigskymountain": 1234567890 } ,
+//     "modes": {"ctf": 1234567890}
+//   }
+// Then, configure your list of indices for OM to look like this:
+//   "indices": [
+//     "pings.us-east",
+//     "modes.ctf",
+//     "modes.teamdm",
+//     "maps.sunsetvalley",
+//     "mmr.rating",
+//   ]
+//
+// For now, OM reads your 'config/matchmaker_config.(json|yaml)' file for a
+// list of indices, which it monitors using the golang module Viper
+// (https://github.com/spf13/viper).
+// In a full deployment, it is expected that you don't manage the config file
+// directly, but instead put the contents of that file into a Kubernetes
+// ConfigMap.  Kubernetes will write those contents to a file inside your
```

Comment:
Managing live indices via kubernetes configmaps makes sense to start, and at least it avoids requiring another data story. I imagine we'll want some kind of live-configiness native to open match that doesn't rely on routing configs through kubernetes, but I think we can come back around to that when we need it.